syntax/intero.vim

```diff
@@ -0,0 +1,51 @@
+"""""""""""
+" Syntax:
+"
+" Custom syntax highlighting for GHCi in the embeded terminal window.
+"""""""""""
+
+" Because we have some regex that use non-ASCII characters
+scriptencoding utf-8
+
+syntax include @haskell syntax/haskell.vim
+
+" Try to distinguish between GHC output and GHCi prompt lines, so that we can
+" enable Haskell highlighting on the GHCi prompts.
+exe 'syntax region GhciLine matchgroup=GhciLinePrompt start=/^'.g:intero_prompt_regex.'/ matchgroup=NONE end=/\n/ oneline keepend contains=@haskell,GhciCommand,GhciLoadCommand'
+syntax match GhciCommand /:[a-z-]\+/ contained
+syntax match GhciLoadCommand /:l.*/ contained contains=GhciCommand
+
+" ----- GHC output ----------------------------------------------------------
+" Some rather ad-hoc regexes for GHC output.
+
+" For when the modules are loaded / reloaded by GHCi.
+syntax region GhciSteps start=/^\[\s*\d\+ of \s*\d\+\]/ end=/)/
+syntax region GhciOk start=/^Ok, modules loaded:/ end=/\.$/
+
+" Experimentally, it seems like GHC outputs "expected/actual" results in one
+" of two formats. Ideally, we'd have match group names describing when one
+" format isused over another.
+syntax region GhciExpectedActual1 start=/\(expected\|actual\) type/ end=/’/ keepend contains=GhciQuotedType
+syntax region GhciQuotedType start=/‘/ end=/’/ contained keepend
+syntax region GhciExpectedActual2  matchgroup=Constant start=/\(Expected\|Actual\) type: / end=/$/
+
+syntax match GhciCause /^\s*• \(Probable\|Possible\) cause: .*$/
+
+
+" ----- Default highlight groups --------------------------------------------
+" Used only if the user's colorscheme doesn't provide its own defaults.
+" In particular, you can configure these for yourself in your vimrc!
+
+" We ignore the prompt GHCi prompts without ANSI color codes don't pick up
+" styles from the underlying Haskell syntax groups.
+hi def link GhciLinePrompt Ignore
```

Comment:
You might think "huh, this might be a neat way to highlight the GHCi prompt without having to use ANSI characters in your prompt!"

I tried this (removing the ANSI colors from my prompt and changing `Ignore` to some other group, like `Type`) and it didn't quite work work. The prompt end up with no colors until I typed a character on that line.

I'm not quite sure where that behavior is coming from. I feel like it should highlight the prompt immediately.